tdms/src/tdms_iterator.cpp

```diff
@@ -6087,28 +6087,38 @@ if(runmode == rm_complete && (nvertices>0) )
   }
   
   //now find the maximum absolute value of residual field in the grid
-  for(k=0;k<(K_tot+1);k++)
-    for(j=0;j<(J_tot+1);j++)
-      for(i=0;i<(I_tot+1);i++){
-	tempfield = fabs(Exy[k][j][i]+Exz[k][j][i]);
-	if( maxfield < tempfield )
-	  maxfield = tempfield;
-	tempfield = fabs(Eyx[k][j][i]+Eyz[k][j][i]);
-	if( maxfield < tempfield )
-	  maxfield = tempfield;
-	tempfield = fabs(Ezx[k][j][i]+Ezy[k][j][i]);	
-	if( maxfield < tempfield )
-	  maxfield = tempfield;
-	tempfield = fabs(Hxy[k][j][i]+Hxz[k][j][i]);	
-	if( maxfield < tempfield )
-	  maxfield = tempfield;
-	tempfield = fabs(Hyx[k][j][i]+Hyz[k][j][i]);	
-	if( maxfield < tempfield )
-	  maxfield = tempfield;
-	tempfield = fabs(Hzx[k][j][i]+Hzy[k][j][i]);	
-	if( maxfield < tempfield )
-	  maxfield = tempfield;
+  // after resetting the maxfield value calculated in the main loop
+  maxfield = 0.0;

```

Comment:
Good pickup, this needs to be here to be correct under all circumstances. In some cases the code will behave the same since maxfield is reset to 0 in the main loop, but only on certain iterations.